utils/env_preferences/src/error.rs

```diff
@@ -49,3 +56,31 @@ impl From<FromVecWithNulError> for RetrievalError {
         Self::FromVecWithNulError(input)
     }
 }
+
+/// An error encountered while either retrieving or parsing a system locale
+#[derive(Display, Debug)]
+pub enum LocaleError {
+    #[displaydoc("Unable to retrieve locales: {0:?}")]
+    Retrieval(RetrievalError),
+    #[displaydoc("Unable to parse locale: {0}")]
+    Parse(ParseError),
+}
+
+#[cfg(any(feature = "parse_posix", target_os = "linux"))]
+impl From<crate::parse::posix::PosixParseError> for LocaleError {
+    fn from(value: crate::parse::posix::PosixParseError) -> Self {
+        Self::Retrieval(RetrievalError::Posix(value))
```

Comment:
why does a `PosixParseError` become a `RetrievalError` instead of a `ParseError`?

Reply:
Good point, this was not very clean and has been resolved :)